src/main/java/org/weakref/jmx/ManagedClass.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright 2022 Matt Stephenson
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.weakref.jmx;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.lang.ref.WeakReference;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import static org.weakref.jmx.ReflectionUtils.getAttributeName;
+import static org.weakref.jmx.ReflectionUtils.isGetter;
+
+public class ManagedClass
+{
+    private final WeakReference<Object> target;
+    private final Map<String, ManagedClass> children;
+    private final Map<String, ManagedAttribute> attributes;
+
+    private ManagedClass(Object target, Map<String, ManagedClass> children, Map<String, ManagedAttribute> attributes)
+    {
+        this.target = new WeakReference<>(target);
+        this.children = children;
+        this.attributes = attributes;
+    }
+
+    public static ManagedClass fromExportedObject(Object target)
+    {
+        Preconditions.checkNotNull(target, "target is null");
```

Comment:
This is not really need. The usages below will cause an NPE anyway, and in Java 14+ it will produce a better error message.